core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/PluginTypeChecker.scala

```diff
@@ -89,20 +94,39 @@ class PluginTypeChecker(platform: String = "onprem") extends Logging {
   def getSupportedExprs: Map[String, String] = supportedExprs
 
   private def readOperatorsScore: Map[String, Double] = {
-    val file = platform match {
-      // if no GPU specified, then default to dataproc-t4 for backward compatibility
-      case "dataproc-t4" | "dataproc" => OPERATORS_SCORE_FILE_DATAPROC_T4
-      case "dataproc-l4" => OPERATORS_SCORE_FILE_DATAPROC_L4
-      // if no GPU specified, then default to emr-t4 for backward compatibility
-      case "emr-t4" | "emr" => OPERATORS_SCORE_FILE_EMR_T4
-      case "emr-a10" => OPERATORS_SCORE_FILE_EMR_A10
-      case "databricks-aws" => OPERATORS_SCORE_FILE_DATABRICKS_AWS
-      case "databricks-azure" => OPERATORS_SCORE_FILE_DATABRICKS_AZURE
-      case _ => OPERATORS_SCORE_FILE_ONPREM
+
+    try {
+      val source = speedupFactorFile match {
+        case None =>
+          logInfo(s"Reading operators scores with platform: $platform")
+          val file = platform match {
+            // if no GPU specified, then default to dataproc-t4 for backward compatibility
+            case "dataproc-t4" | "dataproc" => OPERATORS_SCORE_FILE_DATAPROC_T4
+            case "dataproc-l4" => OPERATORS_SCORE_FILE_DATAPROC_L4
+            // if no GPU specified, then default to emr-t4 for backward compatibility
+            case "emr-t4" | "emr" => OPERATORS_SCORE_FILE_EMR_T4
+            case "emr-a10" => OPERATORS_SCORE_FILE_EMR_A10
+            case "databricks-aws" => OPERATORS_SCORE_FILE_DATABRICKS_AWS
+            case "databricks-azure" => OPERATORS_SCORE_FILE_DATABRICKS_AZURE
+            case _ => OPERATORS_SCORE_FILE_ONPREM
+          }
+          Source.fromResource(file)
+        case Some(file) =>
+          logInfo(s"Reading operators scores from custom speedup factor file: $file")
+          val path = new Path(file)
+          val fs = FileSystem.get(path.toUri, new Configuration())
+          new BufferedSource(fs.open(path))
+      }
+      readSupportedOperators(source, "score").map(x => (x._1, x._2.toDouble))
+    } catch {
+      case e: Exception => speedupFactorFile match {
```

Comment:
```suggestion
      case NonFatal(e) =>
```
It is better to use NonFatal to avoid catching all kinds of exceptions such as OOM

Reply:
Thank you! Changes updated.